### DISCUSSION TITLE
how i can do these tests with TestRedis.php with php-cgi?

### DISCUSSION THREAD

**Discussion by osevan** [phpredis/phpredis#2506] [General]
root@rac:~/compile# /opt/php-8.3/bin/php-cgi /root/compile/test/php-8.3.7/ext/phpredis/tests/TestRedis.php --test get
Status: 500 Internal Server Error
Content-type: text/html; charset=UTF-8

root@rac:~/compile# /root/compile/test/php-8.3.7/sapi/cgi/php-cgi -T1 /root/compile/test/php-8.3.7/ext/phpredis/tests/TestRedis.php --class Redis
Status: 500 Internal Server Error
Content-type: text/html; charset=UTF-8

Elapsed time: 0.000524 sec

With php binary itself , tests passed successfully 

root@rl:~/compile# /opt/php-8.3/bin/php /root/compile/test/php-8.3.7/ext/phpredis/tests/TestRedis.php --test get
Note: these tests might take up to a minute. Don't worry :-)
Using PHP version 8.3.7 (64 bits)
Testing class Redis_Test
testGetSet       [PASSED]
testGetLastError [PASSED]

**Comment by michael-grunder** [comment]
The tests aren't really meant to be run via `php-cgi` so you can't.  You're probably getting pathing issues trying to require test suite files.

Edit:  They can actually be made to run with a couple small changes.

**Comment by michael-grunder** [comment]
This should be fixed in #2508 so you should be able to run them now.

**Comment by osevan** [comment]
https://github.com/phpredis/phpredis/issues/2507#issuecomment-2175913628
